### PR TITLE
[JENKINS-72998] call refresh on the Extension providers again

### PR DESCRIPTION
**Pull request by jtnord**
In the case the `ExtensionFinder` is itself extensible we refresh it again to catch extensions that would be discovered by a newly installed extension.

This seems a little overzerlous, as there is only the variant plugin that I know of, however calling `Jenkins.get().refreshExtensions()` from any point in the `variant` plugin would seem to be a little bit dangerous as it would call back into a place where the reactor is already calling back from.

See [JENKINS-72998](https://issues.jenkins.io/browse/JENKINS-72998).

### Testing done

started a clean jenkins without any plugins
installed ssh-slaves, ssh-credentails & credentials
Attempted to create a new SSH based slave with a credential

validated credentials are not listed before this change and they are after
see https://github.com/jenkinsci/acceptance-test-harness/issues/1526

[incremental build tested with ATH](https://github.com/jenkinsci/acceptance-test-harness/pull/1528) 

### Proposed changelog entries

- JENKINS-72998, If the `variant` plugin is installed at the same time as a plugin that has an `OptionalExtension` then these extensions would not be correctly discovered, until the next scan for new `Extensions`.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by jglick — APPROVED** [review]
:see_no_evil:

Is it possible to mark the `variant` plugin as not supporting hot reload, as a less risky change?

**Comment by jtnord** [comment]
> :see_no_evil:
> 
> Is it possible to mark the `variant` plugin as not supporting hot reload, as a less risky change?

It's possible (it's just code!) but it is a really bad ux as [`ssh-slaves`] (https://github.com/jenkinsci/jenkins/blob/b0baebaf880047ac32428fd58abf62a20e6f9a54/core/src/main/resources/jenkins/install/platform-plugins.json#L76) is a suggested plugin in `platform-plugins` and this would mean the default install flow would now require a restart.

**Comment by jtnord** [comment]
https://ci.jenkins.io/job/Core/job/acceptance-test-harness/job/PR-1528/2/testReport/ shows 2 consistend new failures with this PR.. investigating.

**Comment by jtnord** [comment]
> https://ci.jenkins.io/job/Core/job/acceptance-test-harness/job/PR-1528/2/testReport/ shows 2 consistend new failures with this PR.. investigating.

fails when run with latest master also (but not the last release) so is unrelated to this change

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
